### PR TITLE
doc improvement: wifi: Update memory footprint tables for 2.9

### PR DESCRIPTION
**Pull request by richabp**
Updated Wi-Fi memory footprint tables for the 2.9.0 release. Added memory utilization for 54H and 54L samples.

SHEL-3302

### CODE REVIEW

**Review by umapraseeda — APPROVED** [review]

File: doc/nrf/protocols/wifi/station_mode/mem_requirements_sta.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@ Footprint values are provided in kilobytes (KB).
       +--------------------------------------+-------------+-------------------------------------------+-----------------------------+----------------------+---------------------------------+--------------------+----------------------+
       | Sample                               |   Total ROM |   Wi-Fi driver ROM                        |          nRF70 FW patch ROM |   WPA supplicant ROM |   Total RAM (incl. static heap) |   Wi-Fi driver RAM |   WPA supplicant RAM |
       +======================================+=============+===========================================+=============================+======================+=================================+====================+======================+
-      | :ref:`Station <wifi_station_sample>` |         531 |                                         3 |                          74 |                  187 |                             201 |                127 |                   15 |
+      | :ref:`Station <wifi_station_sample>` |         531 |                                         3 |                          73 |                  187 |                             201 |                127 |                   15 |
```
[the remark this answers is not in the document]
Comment (richabp) [reply doc/nrf/protocols/wifi/station_mode/mem_requirements_sta.rst:21]: Not mentioned here: https://[jenkins-ncs.nordicsemi.no/job/latest/job/test-sdk-wifi-memory/job/master/112/artifact/MEMORY_INFO.LOG](https://jenkins-ncs.nordicsemi.no/job/latest/job/test-sdk-wifi-memory/job/master/112/artifact/MEMORY_INFO.LOG).

cc-@krga2022 @visinor

File: doc/nrf/protocols/wifi/scan_mode/mem_requirements_scan.rst [hunk 2]
```diff
@@ -18,7 +18,7 @@ Footprint values are provided in kilobytes (KB).
       +--------------------------------------+-------------+-------------------------------------------+----------------------------+----------------------+---------------------------------+--------------------+----------------------+
       | Sample                               |   Total ROM |   Wi-Fi driver ROM                        |         nRF70 FW patch ROM |   WPA supplicant ROM |   Total RAM (incl. static heap) |   Wi-Fi driver RAM |   WPA supplicant RAM |
       +======================================+=============+===========================================+============================+======================+=================================+====================+======================+
-      | :ref:`Scan <wifi_scan_sample>`       |         150 |                                         1 |                         28 |                    0 |                              64 |                 34 |                    0 |
+      | :ref:`Scan <wifi_scan_sample>`       |         151 |                                         1 |                         28 |                    0 |                              65 |                 34 |                    0 |
```
[the remark this answers is not in the document]
Comment (richabp) [reply doc/nrf/protocols/wifi/scan_mode/mem_requirements_scan.rst:21]: Not mentioned here: https://[jenkins-ncs.nordicsemi.no/job/latest/job/test-sdk-wifi-memory/job/master/112/artifact/MEMORY_INFO.LOG](https://jenkins-ncs.nordicsemi.no/job/latest/job/test-sdk-wifi-memory/job/master/112/artifact/MEMORY_INFO.LOG).

cc-@krga2022 @visinor

**Review by krga2022 — APPROVED** [review]

File: doc/nrf/protocols/wifi/station_mode/mem_requirements_sta.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (richabp) [reply doc/nrf/protocols/wifi/station_mode/mem_requirements_sta.rst]: Updated as per https://[jenkins-ncs.nordicsemi.no/job/latest/job/test-sdk-wifi-memory/job/v2.9-branch/2/artifact/MEMORY_INFO.LOG](https://jenkins-ncs.nordicsemi.no/job/latest/job/test-sdk-wifi-memory/job/v2.9-branch/2/artifact/MEMORY_INFO.LOG). Please check.

File: doc/nrf/protocols/wifi/scan_mode/mem_requirements_scan.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (richabp) [reply doc/nrf/protocols/wifi/scan_mode/mem_requirements_scan.rst]: Updated as per https://[jenkins-ncs.nordicsemi.no/job/latest/job/test-sdk-wifi-memory/job/v2.9-branch/2/artifact/MEMORY_INFO.LOG](https://jenkins-ncs.nordicsemi.no/job/latest/job/test-sdk-wifi-memory/job/v2.9-branch/2/artifact/MEMORY_INFO.LOG). Please check.

**Review by krish2718 — APPROVED** [review]

**Pull request merged**